tiledb/sm/tile/tile.h

```diff
@@ -191,36 +166,26 @@ class Tile {
       uint64_t cell_size,
       unsigned int dim_num);
 
-  /** Advances the buffer offset. */
-  void advance_offset(uint64_t nbytes);
-
   /** Returns the internal buffer. */
-  inline Buffer* buffer() const {
-    return buffer_;
+  inline void* data() const {
+    return data_.get();
   }
 
-  /** Returns the cell size. */
-  inline uint64_t cell_size() const {
-    return cell_size_;
-  }
+  /** Clears the internal buffer. */
+  void clear_data();
 
   /**
-   * Returns a shallow or deep copy of this Tile.
+   * Allocate the internal buffer.
    *
-   * @param deep_copy If true, a deep copy is performed, including potentially
-   *    memcpying the underlying Buffer. If false, a shallow copy is performed,
-   *    which sets the clone's Buffer equal to Tile's buffer pointer.
-   * @return New Tile
-   */
-  Tile clone(bool deep_copy) const;
-
-  /**
-   * Sets `owns_buffer_` to `false` and thus will not destroy the buffer
-   * in the destructor.
+   * @param size New size.
+   * @return Status.
    */
-  void disown_buff();
+  Status alloc_data(uint64_t size);
```

Comment:
I don't think this function should exist; it's a C.41 issue. The data should be allocated in the constructor, not externally. What's the rationale for having this? Is it to support the default constructor `Tile()`?

I did a quick search on usages, and (outside of tests) there aren't very many uses of the default constructor left. Only 7 lines of code for `Tile()`, and 1 for `WriterTile()` (which uses `Tile()`). Is it time to eliminate it? If not in this PR, in one shortly to follow?